dask_expr/groupby.py

```diff
@@ -0,0 +1,385 @@
+import functools
+
+import numpy as np
+from dask.dataframe.core import (
+    _concat,
+    is_dataframe_like,
+    is_series_like,
+    make_meta,
+    meta_nonempty,
+    no_default,
+)
+from dask.dataframe.groupby import (
+    _agg_finalize,
+    _apply_chunk,
+    _build_agg_args,
+    _determine_levels,
+    _groupby_aggregate,
+    _groupby_apply_funcs,
+    _normalize_spec,
+)
+from dask.utils import M
+
+from dask_expr.collection import DataFrame, new_collection
+from dask_expr.reductions import ApplyConcatApply
+
+
+def _as_dict(key, value):
+    # Utility to convert a single kwarg to a dict.
+    # The dict will be empty if the value is None
+    return {} if value is None else {key: value}
+
+
+###
+### Groupby-aggregation expressions
+###
+
+
+class BaseAggregation(ApplyConcatApply):
+    """Groupby-aggregation base class
+
+    This class does very little besides provide a base
+    class for `SingleAggregation` and `GroupbyAggregation`.
+
+    See Also
+    --------
+    SingleAggregation
+    GroupbyAggregation
+    """
+
+    def _divisions(self):
+        return (None, None)
```

Comment:
If this does very little maybe it doesn't make sense to keep it around?  Does it serve any purpose other than avoiding repeating `_divisions`?  If so, this might be overly-DRY.